nemo_skills/inference/generate.py

```diff
@@ -502,7 +519,14 @@ async def process_single_datapoint(self, data_point, all_data):
             if self.cfg.override_max_code_executions and self.cfg.total_code_executions_in_prompt is not None:
                 generation_params["max_code_executions"] = data_point["total_code_executions"]
 
-        return await self.llm.generate_async(**generation_params)
+        result = await self.llm.generate_async(**generation_params)
+
+        # Apply evaluation hook if configured
```

Comment:
could you put this into another method and call right after process_dingle_datapoint? This will make it easier to reuse as process_single_datapoint is most commonly overriden